src/remote/index.js

```diff
@@ -99,9 +102,33 @@ export default class Remote {
       .catch(callback)
   }
 
-  // FIXME: Temporary stub so we can do some acceptance testing on file upload
+  async overwriteFileAsync (doc: Metadata, old: Metadata): Promise<RemoteDoc> {
+    const stream = await this.other.createReadStreamAsync(doc)
+    const updated = await this.remoteCozy.updateFileById(doc.remote._id, stream, {
+      contentType: doc.mime,
+      checksum: doc.checksum,
+      lastModifiedDate: new Date(doc.lastModification)
+    })
+
+    doc.remote._rev = updated._rev
+
+    return jsonApiToRemoteDoc(updated)
+  }
+
+  overwriteFile (doc: Metadata, old: Metadata, callback: Callback) {
+    this.overwriteFileAsync(doc, old)
+      .then(updated => callback(null, updated))
+      .catch(callback)
```

Comment:
> Isn't this already handled by the catch call on line 121?

I don't think so. `aPromise.catch` only catch a rejected state for the promise, but not the standard thrown exception. And I think await throw such an exception when its promise is rejected.